osvr/RenderKit/RenderManagerBase.cpp

```diff
@@ -1704,18 +1737,179 @@ namespace renderkit {
+      // duration of a full-screen scan-out.  If we are scanning out
+      // from left to right or right to left, divide this by the number
+      // of eyes per display to find the per-eye scan-out duration.
+      RenderTimingInfo timing;
+      if (!GetTimingInfo(0, timing)) {
+        // If we have no timing information, then we have nothing to use
+        // to predict so we return the do-nothing result.
+        return ret;
+      }
+      double screenScanTime = (timing.hardwareDisplayInterval.seconds +
+        timing.hardwareDisplayInterval.microseconds / 1e6);
+      if (edgeUp % 2 == 1) {
+        screenScanTime /= GetNumEyesPerDisplay();
+      }
+
+      // Find out the pose velocity information, if available.
+      // Set the valid flags to false so that if to call to get
+      // velocity fails, we will not try and use the info.
+      OSVR_TimeValue timestamp;
+      OSVR_VelocityState vel;
+      vel.linearVelocityValid = false;
+      vel.angularVelocityValid = false;
+      if (osvrGetVelocityState(m_roomFromHeadInterface, &timestamp, &vel) != OSVR_RETURN_SUCCESS) {
+        // No velocity information available, so we return the do-nothing result.
+        return ret;
+      }
+
+      // Convert the incremental orientation change in world space back
+      // into (local) head space by transforming it by the inverse of the
+      // current head pose.
+      //  Handle a non-Identity room-from-world transform in the OSVR-Core
+      // room-to-world transform (as opposed to the RenderManager one, which is
+      // already handled because we apply that transformation ourselves).  We
+      // do this by getting and applying the room-to-world transform from Core
+      // here.  Again, we can ignore the RenderManager room-to-world that was
+      // passed in as RenderParam because all of our differential transform
+      // work here takes place below it.
+      OSVR_PoseState pose;
+      if (osvrGetPoseState(m_roomFromHeadInterface, &timestamp, &pose) != OSVR_RETURN_SUCCESS) {
+        // No pose information available, so we return the do-nothing result.
+        return ret;
+      }
+      osvr::common::Transform xform(ei::map(pose).matrix(),
+        ei::map(pose).matrix().inverse());
+      xform.transform(m_context->getRoomToWorldTransform());
+      Eigen::Quaterniond localRot = xform.transformDerivative(
+        ei::map(vel.angularVelocity.incrementalRotation));
+
+      // Turn incremental rotation into Euler rotation rates in radians/second.
+      // Do this by converting the Quaternion into Euler and then dividing by the
+      // delta time.  We do this twice, once with the X axis being defined as the
+      // last axis to be rotated around and once with the Y axis as the last. (The
+      // last axis is the first listed in right-to-left multiplication.)  If we
+      // use the same Euler set for more than one angle, sometimes we get flips by
+      // Pi around the axes.
```

Comment:
There is always another way to do the math to get the same result.  I did this the way it made sense to me since I was straining to keep track of all that was happening at the time.  Tested and working as is.  Not opposed to someone modifying it to use another approach and then verifying that it still works afterwards.